File: Coding Ninjas/Easy/Isomorphic Strings.py
def areIsomorphic( str1: str, str2: str ) -> bool:
    n = len( str1 )
    
    if n != len( str2 ):
        return 0
    
    d, e = {}, {}
    
    for i in range( len( str1 ) ):
        if str1[i] not in d:
            if str2[i] in e:
                return 0
            
            d[ str1[i] ] = str2[i]
            e[ str2[i] ] = str1[i]
        elif d[ str1[i] ] != str2[i]:
            return 0
            
    return 1

File: Coding Ninjas/Easy/Isomorphic Strings.java
import java.util.*;

public class Solution
{
    
    public static boolean areIsomorphic
    ( String str1, String str2 )
    {
        if ( str1.length() != str2.length() )
        {
            return false;
        }

        HashMap< Character, Character > map =
            new HashMap<>();

        for ( int i = 0; i < str1.length(); i++ )
        {
            if ( map.containsKey( str1.charAt( i ) ) )
            {
                if ( map.get( str1.charAt( i ) ) != str2.charAt( i ) )
                {
                    return false;
                }
            }
            else if ( map.containsValue( str2.charAt( i ) ) )
            {
                return false;
            }
            else
            {
                map.put( str1.charAt( i ), str2.charAt( i ) );
            }
        }

        return true;
    }

}
